src/mtd.c

```diff
@@ -889,6 +889,11 @@ struct mtd_data *mtd_open(const struct mtd_config *cfg, int flags)
 			goto out;
 		}
 
+		/* override MTD size */
+		if (md->cfg.chip_0_size) {
+			miu->size = md->cfg.chip_0_size;
+		}
+
```

Comment:
I think this will apply for chip 0 and 1. It should be rather easy to actually make it chip 0/1 compatible, just check for i as it is done above with chip_[0|1]_device_path, e.g.

    if (i == 0 && md->cfg.chip_0_size)
        miu->size = md->cfg.chip_0_size;
    else if (i == 1 && md->cfg.chip_1_size)
        miu->size = md->cfg.chip_1_size;